File: Search.SearchService/Internal/CacheRecord.cs
using System;

namespace Search.SearchService.Internal
{
    public class CacheRecord
    {
        public SearchResponse Response { get; }
        public TimeSpan CreationTime { get; set; }

        public CacheRecord(SearchResponse response, TimeSpan creationTime)
        {
            Response = response;
            CreationTime = creationTime;
        }
    }
}

File: FSharpSearchService/CacheRecord.fs
namespace FSharpSearchService

open System

type CacheRecord()=

    [<DefaultValue>]
    val mutable public Response :SearchResponse
    member this.response
        with get()=this.Response
     
    [<DefaultValue>]
    val mutable public CreationTime:TimeSpan
    member this.creationTime
        with get()=this.CreationTime
        and set(value)=this.CreationTime<-value